src/append_only_data.rs

```diff
@@ -242,6 +242,18 @@ pub struct Owner {
     pub permissions_index: u64,
 }
 
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize, Default, Debug)]
+pub struct Entry {
+    pub key: Vec<u8>,
+    pub value: Vec<u8>,
+}
+
+impl Entry {
+    pub fn new(key: Vec<u8>, value: Vec<u8>) -> Self {
```

Comment:
Looks good! I just don't understand why we want `Entry::new`, the fields are public so you can just do `Entry { key: X, value: Y }` which is a bit more explicit and not much more verbose.

Reply:
Just makes it slightly more ergonomic to use.  For example, see all the changes in the tests which would have been a bit more awkward without the constructor.  E.g.
```rust
let entries = vec![
    Entry {
        key: b"KEY1".to_vec(),
        value: b"VALUE1".to_vec(),
    },
    Entry {
        key: b"KEY2".to_vec(),
        value: b"VALUE2".to_vec(),
    },
    Entry {
        key: b"KEY1".to_vec(),
        value: b"VALUE1".to_vec(),
    },
];
```
vs
```rust
let entries = vec![
    Entry::new(b"KEY1".to_vec(), b"VALUE1".to_vec()),
    Entry::new(b"KEY2".to_vec(), b"VALUE2".to_vec()),
    Entry::new(b"KEY1".to_vec(), b"VALUE1".to_vec()),
];
```